### PR TITLE
[JENKINS-64151] Set a default file size rotation of AsyncPeriodicWork…

### PR DESCRIPTION
**Pull request by Dohbedoh**
… / AsyncAperiodicWork task logs

See [JENKINS-64151](https://issues.jenkins.io/browse/JENKINS-64151).

Be preventive of intensive task logging, that could severely impact disk usage otherwise, by setting a default size rotation. Chose 10 MB as a default as 10 MB seems like already quite a lot for a single periodic task to log in a file in a day. This is also just above what a periodic task that run **every seconds** (that is extremely aggressive) would log by default (about 7MB and 172800 lines):

```
Started at Mon Nov 14 00:00:01 UTC 2022
Finished at Mon Nov 14 00:00:01 UTC 2022. 0ms
[....]
```

### Testing done

* Started Jenkins fresh
* Run the `FingerprintCleanupThread` in script console with `hudson.model.FingerprintCleanupThread.getInstance().run()` (a core PeriodicWork)
* Fill up the file `$JENKINS_HOME/logs/tasks/Fingerprint cleanup.log` to be just a few bytes above the limit

```
% ls -la $JENKINS_HOME/logs/tasks/
total 20608
drwxr-xr-x  3 myuser  staff        96 14 Nov 15:45 .
drwxr-xr-x  3 myuser  staff        96 14 Nov 15:45 ..
-rw-r--r--  1 myuser  staff  10485868 14 Nov 15:53 Fingerprint cleanup.log
```

* Run the `FingerprintCleanupThread` in script console with `hudson.model.FingerprintCleanupThread.getInstance().run()`
* The file should be rotated:

```
% ls -la $JENKINS_HOME/logs/tasks/
total 20616
drwxr-xr-x  4 myuser  staff       128 14 Nov 15:53 .
drwxr-xr-x  3 myuser  staff        96 14 Nov 15:45 ..
-rw-r--r--  1 myuser  staff       227 14 Nov 15:53 Fingerprint cleanup.log
-rw-r--r--  1 myuser  staff  10485868 14 Nov 15:53 Fingerprint cleanup.log.1
```

### Proposed changelog entries

- Set default file size rotation of AsyncPeriodicWork / AsyncAperiodicWork task logs.

### Proposed upgrade guidelines

- Periodic task log files are now rotated based on the file size with a default size of 10 MB. They were previously not rotated based on the file size. To revert to the previous behavior and disable the file size rotation, the system properties [hudson.model.AsyncAperiodicWork.logRotateSize=-1](https://www.jenkins.io/doc/book/managing/system-properties/#hudson-model-asyncaperiodicwork-logrotatesize) and [hudson.model.AsyncPeriodicWork.logRotateSize=-1](https://www.jenkins.io/doc/book/managing/system-properties/#hudson-model-asyncperiodicwork-logrotatesize) can be used.

### Submitter checklist

- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

NA

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7376"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
